core/tests/ledger_cleanup.rs

```diff
@@ -439,12 +451,18 @@ mod tests {
             }
         }
         insert_exit.store(true, Ordering::Relaxed);
-        insert_thread.join().unwrap();
 
-        info!(
-            "done {} {} shreds/s",
+        while !insert_threads.is_empty() {
+            if let Some(thread) = insert_threads.pop() {
+                thread.join().unwrap();
+            }
+        }
```

Comment:
Checking if the vec is empty and if `.pop()` is redundant; you can just do something like this
```
        while let Some(thread) = insert_threads.pop() {
            thread.join().unwrap();
        }
```

Reply:
Neat!  Thanks for the suggestion.